File: src/main/java/com/ferreusveritas/exampletrees/blocks/BlockIronLog.java
package com.ferreusveritas.exampletrees.blocks;

import com.ferreusveritas.dynamictrees.DynamicTrees;

import net.minecraft.block.BlockLog;
import net.minecraft.block.properties.IProperty;
import net.minecraft.block.state.BlockStateContainer;
import net.minecraft.block.state.IBlockState;

public class BlockIronLog extends BlockLog {

	public static final String name = "ironlog";
	
	public BlockIronLog() {
		this(name);
		setCreativeTab(DynamicTrees.dynamicTreesTab);
	}
	
	public BlockIronLog(String name) {
        this.setDefaultState(this.blockState.getBaseState().withProperty(LOG_AXIS, BlockLog.EnumAxis.Y));
		setDefaultState(this.blockState.getBaseState());
		setUnlocalizedName(name);
		setRegistryName(name);
	}
	
    @Override
	protected BlockStateContainer createBlockState() {
        return new BlockStateContainer(this, new IProperty[] {LOG_AXIS});
    }
	
    @Override
	public IBlockState getStateFromMeta(int meta) {
    	    	
        IBlockState iblockstate = this.getDefaultState();

        switch (meta & 12) {
            case 0: iblockstate = iblockstate.withProperty(LOG_AXIS, BlockLog.EnumAxis.Y);  break;
            case 4: iblockstate = iblockstate.withProperty(LOG_AXIS, BlockLog.EnumAxis.X);  break;
            case 8: iblockstate = iblockstate.withProperty(LOG_AXIS, BlockLog.EnumAxis.Z);  break;
            default: iblockstate = iblockstate.withProperty(LOG_AXIS, BlockLog.EnumAxis.NONE);
        }
        
        return iblockstate;
    }

    /**
     * Convert the BlockState into the correct metadata value
     */
    @Override
	public int getMetaFromState(IBlockState state) {
        int i = 0;

        switch (state.getValue(LOG_AXIS)) {
            case X:  i |= 4; break;
            case Z:  i |= 8; break;
            case NONE: i |= 12; break;
            default: break;
        }
        
        return i;
    }
}

File: src/main/java/com/ferreusveritas/exampletrees/blocks/BlockIronLog.kt
package com.ferreusveritas.exampletrees.blocks

import com.ferreusveritas.dynamictrees.DynamicTrees
import net.minecraft.block.BlockLog
import net.minecraft.block.state.BlockStateContainer
import net.minecraft.block.state.IBlockState

class BlockIronLog(name: String) : BlockLog() {

    constructor() : this("ironlog") {
        setCreativeTab(DynamicTrees.dynamicTreesTab)
    }

    init {
        this.defaultState = this.blockState.baseState.withProperty(BlockLog.LOG_AXIS, BlockLog.EnumAxis.Y)
        defaultState = this.blockState.baseState
        unlocalizedName = name
        setRegistryName(name)
    }

    override fun createBlockState(): BlockStateContainer {
        return BlockStateContainer(this, BlockLog.LOG_AXIS)
    }

    override fun getStateFromMeta(meta: Int): IBlockState {

        var iblockstate = this.defaultState

        iblockstate = when (meta and 12) {
            0 -> iblockstate.withProperty(BlockLog.LOG_AXIS, BlockLog.EnumAxis.Y)
            4 -> iblockstate.withProperty(BlockLog.LOG_AXIS, BlockLog.EnumAxis.X)
            8 -> iblockstate.withProperty(BlockLog.LOG_AXIS, BlockLog.EnumAxis.Z)
            else -> iblockstate.withProperty(BlockLog.LOG_AXIS, BlockLog.EnumAxis.NONE)
        }

        return iblockstate
    }

    /**
     * Convert the BlockState into the correct metadata value
     */
    override fun getMetaFromState(state: IBlockState): Int {
        var i = 0

        when (state.getValue(BlockLog.LOG_AXIS)) {
            BlockLog.EnumAxis.X -> i = i or 4
            BlockLog.EnumAxis.Z -> i = i or 8
            BlockLog.EnumAxis.NONE -> i = i or 12
            else -> {
            }
        }

        return i
    }

}
